### PR TITLE
Fix #1991 - Additional import options for import_cv2.py

### LINKED ISSUE #1991: Filtering out-of-alphabet characters
As it turns out, the English sets in CommonVoice v2.0 contain out-of-alphabet (of the English 26 letter one plus space plus apostrophe) characters - `[':', 'ú', 'l', "'", 'é', 'á', ';', '’', 'ō', '`&#96;`', '!', ' ', '–', 'ó', 'â', 'ë', 't', 'ñ',]`. While some of them seem to be a case for `validate_label`, others most probably come from borrowed words of other languages. How should we deal with this cases - short term and long term?
As DeepSpeech fails importing csv files containing such characters in transcripts we need a short term solution which could be one or more of:
 - Adding an option for ignoring samples containing out-of-alphabet transcripts to `DeepSpeech.py`.
- Adding a similar option to `import_cv2.py` along with specifying the alphabet file to use.
- Providing a tool to filter samples from CSV files (providing an alphabet) after import.
- Adding this kind of functionality to the voice corpus tool.

What do you vote for?

### PR DESCRIPTION
**Pull request by tilmankamp**

### CODE REVIEW

File: bin/import_cv2.py [hunk 1]
```diff
@@ -137,8 +131,28 @@ def _maybe_convert_wav(mp3_filename, wav_filename):
...
+    parser.add_argument('--normalize', action='store_true', help='Converts diacritic characters to their base ones')
+    parser.set_defaults(normalize=False)
+    params = parser.parse_args()
+
+    audio_dir = params.audio_dir if params.audio_dir else os.path.join(params.tsv_dir, 'clips')
+    alphabet = Alphabet(params.filter_alphabet) if params.filter_alphabet else None
+
+    def label_filter(label):
+        if params.normalize:
+            label = unicodedata.normalize("NFKD", label.strip()) \
```
Comment (kdavis-mozilla) [inline bin/import_cv2.py]: So this is just for English for now.

File: bin/import_cv2.py [hunk 2]
```diff
@@ -137,8 +131,28 @@ def _maybe_convert_wav(mp3_filename, wav_filename):
...
+                .encode("ascii", "ignore") \
+                .decode("ascii", "ignore")
+        label = validate_label(label)
+        if alphabet and label:
+            try:
+                [alphabet.label_from_string(c) for c in label]
+            except KeyError:
+                return None
```
Comment (kdavis-mozilla) [inline bin/import_cv2.py]: nit: Why not set label to None to avoid multiple returns.

File: bin/import_cv2.py [hunk 3]
```diff
@@ -9,49 +9,42 @@
 
 import csv
 import sox
+import argparse
```
Comment (kdavis-mozilla) [inline bin/import_cv2.py]: General question that I'll just put here: What about other languages?

CV2 now has about 15 languages and if I want to import anything other than English, it doesn't look like we support that. Maybe another PR to add non-English support?

File: bin/import_cv2.py (hunk 1 above)
Comment (tilmankamp) [reply bin/import_cv2.py]: This particular option only makes sense for English, yes (as far as I can tell).

File: bin/import_cv2.py (hunk 3 above)
Comment (tilmankamp) [reply bin/import_cv2.py]: One has to specify the according alphabet and it should be fine for most cases. I see the `normalize` option as something hacky that is really optional and only makes sense for English as of now. None of the other changes is specifically targeting English and this PR anyhow doesn't aim for solving all of these problems. It's purpose is to allow filtering samples with unacceptable characters. Common character conversion and filtering should be done in `validate_label` (like casing), while everything else is a matter of providing the right alphabet.

File: bin/import_cv2.py [hunk 4]
```diff
@@ -137,8 +131,28 @@ def _maybe_convert_wav(mp3_filename, wav_filename):
 
 
 if __name__ == "__main__":
-    audio_dir = sys.argv[1]
-    tsv_dir = sys.argv[2]
-    print('Expecting your audio from Common Voice to be in: ', audio_dir)
-    print('Looking for *.tsv files (generated by CorporaCreator) in: ', tsv_dir)
-    _preprocess_data(audio_dir, tsv_dir)
+    parser = argparse.ArgumentParser(description='Import CommonVoice v2.0 corpora')
+    parser.add_argument('tsv_dir', help='Directory containing tsv files')
```
[the remark this answers is not in the document]
Comment (tilmankamp) [reply bin/import_cv2.py]: It is - just tested it.

[the remark this answers is not in the document]
Comment (kdavis-mozilla) [reply bin/import_cv2.py]: Strange, the [required](https://docs.python.org/2/library/argparse.html#required) documentation for the [add_argument()](https://docs.python.org/2/library/argparse.html#the-add-argument-method) method suggests `required=True` is need to make an argument required.

I guess if it is a positional argument, it defaults to not required, but if it is a flag argument it defaults to optional.

File: bin/import_cv2.py [hunk 5]
```diff
@@ -137,8 +131,28 @@ def _maybe_convert_wav(mp3_filename, wav_filename):
...
+    parser.add_argument('--audio_dir', help='Directory containing the audio clips - defaults to "<tsv_dir>/clips"')
+    parser.add_argument('--filter_alphabet', help='Exclude samples with characters not in provided alphabet')
+    parser.add_argument('--normalize', action='store_true', help='Converts diacritic characters to their base ones')
```
[the remark this answers is not in the document]
Comment (tilmankamp) [reply bin/import_cv2.py]: No

File: bin/import_cv2.py (hunk 2 above)
Comment (tilmankamp) [reply bin/import_cv2.py]: Changed

File: bin/import_cv2.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (reuben) [reply bin/import_cv2.py]: Positional arguments are required by default.

File: bin/import_cv2.py [hunk 6]
```diff
@@ -137,8 +131,28 @@ def _maybe_convert_wav(mp3_filename, wav_filename):
...
+    parser.set_defaults(normalize=False)
```
[the remark this answers is not in the document]
Comment (kdavis-mozilla) [reply bin/import_cv2.py]: Is there a reason not to use [default](https://docs.python.org/3/library/argparse.html#default)?

Having the default value and the parameter defined as the same place seems to make more sense from a maintenance point of view.

[the remark this answers is not in the document]
Comment (reuben) [reply bin/import_cv2.py]: FWIW, `action='store_true'` already implies a default of False: https://docs.python.org/3/library/argparse.html#action

[the remark this answers is not in the document]
Comment (kdavis-mozilla) [reply bin/import_cv2.py]: @reuben I agree, which makes the line
```python
parser.set_defaults(normalize=False)
```
unneeded.

[the remark this answers is not in the document]
Comment (tilmankamp) [reply bin/import_cv2.py]: It is already gone.

File: README.md [hunk 7]
```diff
@@ -251,46 +251,35 @@ Please ensure you have the required [CUDA dependency](#cuda-dependency).
...
-
-- `cv-other-train.csv`
-- `cv-other-dev.csv`
-- `cv-other-test.csv`
-
-`cv-invalid.csv` contains all samples that users flagged as invalid.
-
-A sub-directory called `cv_corpus_{version}` contains the mp3 and wav files that were extracted from an archive named `cv_corpus_{version}.tar.gz`.
-All entries in the CSV files refer to their samples by absolute paths. So moving this sub-directory would require another import or tweaking the CSV files accordingly.
+All entries in these CSV files refer to their samples by absolute paths. So moving this sub-directory would require another import or tweaking the CSV files accordingly.
```
Comment (kdavis-mozilla) [inline README.md]: Why does the importer do this and not use relative paths? As far as I know Deep Speech allows for relative paths.

**Pull request merged**